include/libtorrent/udp_socket.hpp

```diff
@@ -99,48 +100,28 @@ namespace libtorrent
 		void set_force_proxy(bool f) { m_force_proxy = f; }
 
 		bool is_closed() const { return m_abort; }
-		tcp::endpoint local_endpoint(error_code& ec) const
-		{
-			udp::endpoint ep = m_ipv4_sock.local_endpoint(ec);
-			return tcp::endpoint(ep.address(), ep.port());
-		}
-
-		void set_buf_size(int s);
+		udp::endpoint local_endpoint(error_code& ec) const
+		{ return m_socket.local_endpoint(ec); }
 
 		typedef udp::socket::receive_buffer_size receive_buffer_size;
 		typedef udp::socket::send_buffer_size send_buffer_size;
 
 		template <class SocketOption>
 		void get_option(SocketOption const& opt, error_code& ec)
 		{
-#if TORRENT_USE_IPV6
-			if (opt.level(udp::v6()) == IPPROTO_IPV6)
-				m_ipv6_sock.get_option(opt, ec);
-			else
-#endif
-				m_ipv4_sock.get_option(opt, ec);
+				m_socket.get_option(opt, ec);
 		}
 
 		template <class SocketOption>
 		void set_option(SocketOption const& opt, error_code& ec)
 		{
-			if (opt.level(udp::v4()) != IPPROTO_IPV6)
-				m_ipv4_sock.set_option(opt, ec);
-#if TORRENT_USE_IPV6
-			if (opt.level(udp::v6()) != IPPROTO_IP)
-				m_ipv6_sock.set_option(opt, ec);
-#endif
```

Comment:
Without this there needs to be some other way to avoid setting dont_fragment on IPv6 sockets. Right now it will assert because there is no DF option in IPv6 (it never fragments).